### ISSUE TITLE
DiamondOperator complains when instantiating/assigning inner types

### ISSUE THREAD

**Issue by carlosmiranda** [teamed/qulice#779]
`DiamondOperator` complains when you instantiate and assign an inner class type. For example, see the following code which is typical when using the [Takes](https://github.com/yegor256/takes) framework:

```java
    @Override
    public Response act(final Request req) throws IOException {
        final RqForm.Smart smart = new RqForm.Smart(new RqForm.Base(req));
        // rest of code
    }
```

Qulice will complain about this:

```
[INFO] Checkstyle: src/test/java/com/xockets/xjm/http/TkAppTest.java[96]: Use diamond operator (DiamondOperatorCheck)
```

Note that `RqForm.Smart` is NOT a generic type, and neither are its parent types `RqForm` or `Request`. There is no way that the diamond operator can be applied here.

**Comment by krzyk** [comment]
@davvd valid bug

**Comment by krzyk** [comment]
@carlosmiranda thanks

**Comment by davvd** [comment]
> @davvd valid bug

@krzyk tag "bug" added

**Comment by davvd** [comment]
@carlosmiranda many thanks for the report, I topped your account for 15 mins, transaction 89309942

**Comment by krzyk** [comment]
@davvd this is urgent

**Comment by davvd** [comment]
> @davvd this is urgent

@krzyk sure, thanks, I added "urgent" label to it

**Comment by davvd** [comment]
@dskalenko this task is for you, keep [these principles](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) in mind, and don't hesitate to ask any technical questions. The budget of this issue is **30 mins**, which is exactly how much will be paid when the task is *done* (see [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) for explanation)

**Comment by dskalenko** [comment]
@davvd I need more time to fix this issue, thanks for understanding

**Comment by davvd** [comment]
> @davvd I need more time to fix this issue, thanks for understanding

@dskalenko sure, take your time

**Comment by dskalenko** [comment]
@davvd here my PR #789

**Comment by davvd** [comment]
> @davvd here my PR #789 

@dskalenko thanks

**Comment by krzyk** [comment]
@dskalenko now you have to ask the issue creator to close it

**Comment by carlosmiranda** [comment]
@dskalenko thanks!

**Comment by krzyk** [comment]
@davvd this is not urgent

**Comment by davvd** [comment]
> @davvd this is not urgent

@krzyk got it, "urgent" tag removed from here

**Comment by davvd** [comment]
@dskalenko thanks, paid, **30 mins** to your account, payment ID is `95473345`, task took 1148 hours and 22 mins to complete... +30 to your rating, your total score is [+60](http://www.netbout.com/b/36376?open=rating)

**Comment by caarlos0** [comment]
@mkordas can you release it?

**Comment by caarlos0** [comment]
@krzyk can you release it plz?

**Comment by krzyk** [comment]
@caarlos0 it was released on 0.17.1
